shepherd_bms/src/eepromdirectory.cpp

```diff
@@ -0,0 +1,82 @@
+#include "eepromdirectory.h"
+
+void eepromInit()
+{   
+    // Initialize EEPROM addresses given data and length
+
+    int i = 1;
+    int offset = 0;
+
+    /* initialize root offset to zero */
+    eeprom_data[0].offset = EEPROM_ROOT_ADDR;
+
+    /* continue through array, setting offsets */
+    while (eeprom_data[i].id != NULL)
+    {
+        offset += eeprom_data[i-1].size;
+        eeprom_data[i].offset = offset;
+        i++;
+    }
+}
+
+int eepromGetIndex(char *key)
```

Comment:
We probably want an inverse of this function too just for usability (i.e. given a offset, what key is at that index)